jax/_src/lax/linalg.py

```diff
@@ -1441,3 +1441,119 @@ def tridiagonal_solve(dl, d, du, b):
     raise ValueError(f'Only f32/f64 are supported, got {t}')
 
   return tridiagonal_solve_p.bind(dl, d, du, b, m=m, n=n, ldb=ldb, t=t)
+
+
+# Schur Decomposition
+
+
+def schur(x,
+          compute_schur_vectors=True,
+          sort_eig_vals=False,
+          select_callable=None):
+  return schur_p.bind(
+      x,
+      compute_schur_vectors=compute_schur_vectors,
+      sort_eig_vals=sort_eig_vals,
+      select_callable=select_callable)
+
+
+def _schur_impl(operand, *, compute_schur_vectors, sort_eig_vals,
+               select_callable):
+  return (xla.apply_primitive(
+      schur_p,
+      operand,
+      compute_schur_vectors=compute_schur_vectors,
+      sort_eig_vals=sort_eig_vals,
+      select_callable=select_callable))
+
+
+def _schur_translation_rule(c, operand, *, compute_schur_vectors, sort_eig_vals):
+  raise NotImplementedError(
+      "Schur decomposition is only implemented on the CPU backend.")
+
+
+def _schur_abstract_eval(operand, *, compute_schur_vectors, sort_eig_vals,
+                        select_callable):
+  if isinstance(operand, ShapedArray):
+    if operand.ndim < 2 or operand.shape[-2] != operand.shape[-1]:
+      raise ValueError("Argument to Schur decomposition must have "
+                       "shape [..., n, n], got shape {}".format(operand.shape))
+
+    batch_dims = operand.shape[:-2]
+    n = operand.shape[-1]
+    dtype = operand.dtype
+    dtype = dtypes.canonicalize_dtype(dtype)
+    T = operand.update(shape=batch_dims + (n, n), dtype=dtype)
+    vs = operand.update(shape=batch_dims + (n, n), dtype=dtype)
+  else:
+    raise NotImplementedError
+
+  output = [T]
+  if compute_schur_vectors:
+    output.append(vs)
+
+  return tuple(output)
+
+
+def _schur_cpu_translation_rule(c, operand, *, compute_schur_vectors,
+                               sort_eig_vals, select_callable):
+  shape = c.get_shape(operand)
+  batch_dims = shape.dimensions()[:-2]
+
+  _cpu_gees = lapack.gees
```

Comment:
Before doing this, perhaps we should check the jaxlib version and raise an informative error if it is too old. As currently written, if the user has too old a version of jaxlib it will be an unadorned AttributeError.

Reply:
I imported the jaxlib version `from jax._src.lib import version as jaxlib_version` and added a condition that raises a NotImplemented error before the reference to `lapack.gees`.